src/with_ladj.jl

```diff
@@ -84,9 +84,19 @@ _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj::Tuple{Any,Real}) = y_with
 function _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj)
     y = map_or_bc(_get_y, y_with_ladj)
     ladj = sum(map_or_bc(_get_ladj, y_with_ladj))
+    #ladj = sum(_get_ladj, y_with_ladj)
```

Comment:
Why is this not the implementation? It would avoid all the allocations. Alternatively, if we want to ensure that pairwise summation is used we could write
```suggestion
    ladj = sum(Broadcast.instantiate(Broadcast.broadcasted(last, y_with_ladj)))
```

Reply:
We could also use `sum(last, y_with_ladj)`, but that didn't help, regarding AD-time.

Reply:
This is expected since the `rrule` for `sum(f, xs)` is not very efficient and computes an array of the primal values and the pullbacks of each entry: https://github.com/JuliaDiff/ChainRules.jl/blob/edf3a1f48fb5c9af01820aeca6ced94d4f97fa1a/src/rulesets/Base/mapreduce.jl#L66-L91

Reply:
I assume that one of the main inefficiencies is that there are no efficient rules defined for `sum(last, ::AbstractArray{<:Tuple})` and `map(first, ::AbstractArray{<:Tuple})` in ChainRules.